src/main/kotlin/org/rust/ide/inspections/RustNamingInspection.kt

```diff
@@ -254,7 +254,8 @@ class RustVariableNamingInspection : RustSnakeCaseNamingInspection("Variable") {
     override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) =
         object : RustElementVisitor() {
             override fun visitPatBinding(el: RustPatBindingElement) {
-                if (el.parent?.parent !is RustParameterElementImpl) {
+                val parent = el.parent?.parent
+                if (parent is RustLetDeclElement || parent is RustPatTupElement) {
```

Comment:
The solution with `PsiTreeUtil.getTopmostParentOfType()` surely works better. Though, I haven't yet gain a good enough understanding of the PSI tree structure to grasp what's happening inside this function or what `rootPattern` should do.